c7n/resources/opsworks.py

```diff
@@ -13,8 +13,34 @@
 # limitations under the License.
 from __future__ import absolute_import, division, print_function, unicode_literals
 
+from botocore.exceptions import ClientError
+
+from c7n.actions import BaseAction
 from c7n.manager import resources
 from c7n.query import QueryResourceManager
+from c7n.utils import local_session, type_schema
+from c7n import utils
+
+
+class StateTransitionFilter(object):
```

Comment:
rather than copying and pasting this from ec2, can you just import that class def and use it here? ie.

from c7n.resources.ec2 import StateTransitionFilter

Reply:
I had to change line 40 from `if i['State']['Name'] in states]` to `if i['Status'] in states]` so I decided to copy over instead of trying to modify the original class to include a parameter to dictate what index to us or some similar workaround. Creating a separate class seemed more appropriate.

Reply:
ah thanks, that makes sense, always helpful to put that as a comment in the code for other readers.

Reply:
Added comment